Combine all styles (#404)
ansys/pyansys-dev-guide: doc/source/all-styles.rst

Style
=====

> In the PyAnsys ecosystem, three key styles—packaging, coding, and documentation—

Reviewer: ```suggestion
In the PyAnsys ecosystem, three key styles —packaging, coding, and documentation—
```

Reviewer 2: @RobPasMue @dipinknair Google dev style guideline says no spaces before dashes. (This is also true in other style guides.) I generally prefer using commas over dashes: "In the PyAnsys ecosystem, three key styles, packaging, coding, and documentation, ...." Vale will throw an error if the spaces appear before the dashes.

Reviewer: Oh, didn't know that!

> In the PyAnsys ecosystem, three key styles — Packaging, Coding, and Documentation 

Reviewer 2: ```suggestion
In the PyAnsys ecosystem, three key styles—packaging, coding, and documentation—
```

serve as foundational pillars to enhance the overall developer experience.

> The Packaging style focuses on creating clear, open-source APIs hosted on GitHub,

Reviewer 2: ```suggestion
Packaging style focuses on creating clear, open source APIs hosted on GitHub,
```
Google dev guide says it is "open source" even when used as a compound adjective.

allowing for reusable packages that can be updated independently of the Ansys release schedule.

> The Coding style adheres to PEP 8 and aligns with the conventions of 
> major data science libraries like numpy, scipy, and pandas, ensuring consistency and readability.
> Lastly, the Documentation style emphasizes the significance of well-documented APIs, 
> offering increased adoption, improved onboarding experiences, and streamlined code maintenance.

Reviewer 2: ```suggestion
- Coding style adheres to PEP 8 and aligns with the conventions of major data science libraries
like NumPy, SciPy, and pandas, ensuring consistency and readability.
- Documentation style emphasizes the significance of well-documented APIs, 
offering increased adoption, an improved on-boarding experiences, and streamlined code maintenance.
```

> All PyAnsys libraries follow the `Google developer documentation style guide
> <https://developers.google.com/style/>`_, incorporating sentence case, 
> active voice, and concise, clear sentences for cohesive and user-friendly documentation.

Reviewer: I think this info shouldn't go here, but in the "doc style" section.
```suggestion
```

Style
=====

In the PyAnsys ecosystem, three key styles—packaging, coding, and documentation—
serve as foundational pillars to enhance the overall developer experience.

- Packaging style focuses on creating clear, open source APIs hosted on GitHub,allowing
  for reusable packages that can be updated independently of the Ansys release schedule.

- Coding style adheres to PEP 8 and aligns with the conventions of major data science libraries
  like NumPy, SciPy, and pandas, ensuring consistency and readability.

- Documentation style emphasizes the significance of well-documented APIs, 
  offering increased adoption, an improved on-boarding experiences, and streamlined code maintenance.

> All PyAnsys libraries follow the Google developer documentation style guide, 

Reviewer 2: Make ``Google developer documentation style guide`` a link: https://developers.google.com/style/

Author: Did you mean the space for the second line of bullet point? 
?

incorporating sentence case, active voice, and concise, clear sentences for 
cohesive and user-friendly documentation.

.. grid:: 3

    .. grid-item-card:: :octicon:`file-directory` Packaging
        :link: packaging/index
        :link-type: doc

        Best practices for distributing Python code.

    .. grid-item-card:: :octicon:`codespaces` Coding
        :link: coding-style/index
        :link-type: doc

        Best practices for writing Python code.

    .. grid-item-card:: :octicon:`pencil` Documentation
        :link: doc-style/index
        :link-type: doc

>         Best practices for writing documentation.

Reviewer 2: ```suggestion
        Best practices for writing PyAnsys library documentation.
```
